src/component/autocomplete.js

```diff
@@ -211,49 +217,78 @@ export class AutoCompleteCustomElement {
    * @param {string} newValue
    * @param {string} oldValue
    *
-   * @returns {void}
+   * @returns {Promise}
    */
   searchChanged(newValue, oldValue) {
     if (!this.shouldPerformRequest()) {
       this.results = [];
 
-      return;
+      return Promise.resolve();
+    }
+
+    // when resource is not defined it will not perform a request. Instead it
+    // will search for the first items that pass the predicate
+    if (this.items) {
+      this.results = this.sort(this.filter(this.items));
+
+      return Promise.resolve();
     }
 
-    return this.findResults(this.searchQuery(this.search)).then(results => {
+    let promise = this.findResults(this.searchQuery(this.search)).then(results => {
```

Comment:
Why not use: `this.lastFindPromise = ....`?

It's a sync process as far as I can see, so the assignment is basically the same, isn't it?